contracts/nft/erc1155m/ERC1155MInitializableV1_0_0.sol

```diff
@@ -0,0 +1,580 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {
+    ERC1155SupplyUpgradeable,
+    ERC1155Upgradeable
+} from "@openzeppelin/contracts-upgradeable/token/ERC1155/extensions/ERC1155SupplyUpgradeable.sol";
+import {ERC2981} from "solady/src/tokens/ERC2981.sol";
+import {Ownable} from "solady/src/auth/Ownable.sol";
+import {ReentrancyGuard} from "solady/src/utils/ReentrancyGuard.sol";
+import {MerkleProofLib} from "solady/src/utils/MerkleProofLib.sol";
+import {SafeTransferLib} from "solady/src/utils/SafeTransferLib.sol";
+
+import {MintStageInfo1155} from "../../common/Structs.sol";
+import {MINT_FEE_RECEIVER} from "../../utils/Constants.sol";
+import {IERC1155M} from "./interfaces/IERC1155M.sol";
+import {ERC1155MStorage} from "./ERC1155MStorage.sol";
+import {Cosignable} from "../../common/Cosignable.sol";
+import {AuthorizedMinterControl} from "../../common/AuthorizedMinterControl.sol";
+
+/// @title ERC1155MInitializableV1_0_0
+/// @notice An upgradeable ERC1155 contract with multi-stage minting, royalties, and authorized minters
```

Comment:
Suggest:
`@notice An upgradeable ERC1155 contract `
`@notice An initializable ERC1155 contract`

Unless this implementation contract is intended to be fully upgradeable.